csodiaq/identifier/targetedReanalysisFunctions.py

```diff
@@ -0,0 +1,232 @@
+from csodiaq.utils import format_protein_list_to_string
+import pandas as pd
+import numpy as np
+
+
+def calculate_mz_of_heavy_version_of_peptide(peptide, lightMz, z):
+    lightAndHeavyLysKMassDiff = 8.014199
+    lightAndHeavyArgRMassDiff = 10.00827
+
+    numLysK = peptide.count("K")
+    numArgR = peptide.count("R")
+
+    return (
+        lightMz
+        + (numLysK * lightAndHeavyLysKMassDiff) / z
+        + (numArgR * lightAndHeavyArgRMassDiff) / z
+    )
+
+
+def filter_to_only_keep_peptides_with_possibly_heavy_K_or_R_terminal_residue(fdrDf):
+    return fdrDf[
+        fdrDf["peptide"].str.endswith("R") | fdrDf["peptide"].str.endswith("K")
+    ].reset_index(drop=True)
+
+
+def filter_to_only_keep_top_peptides_unique_to_protein(fdrDf, maxPeptidesPerProtein):
+    fdrDf = (
+        fdrDf[fdrDf["uniquePeptide"] == 1]
+        .sort_values("ionCount", ascending=False)
+        .reset_index(drop=True)
+    )
+    return (
+        fdrDf.groupby(["leadingProtein"])
+        .head(maxPeptidesPerProtein)
+        .reset_index(drop=True)
+    )
+
+
+def calculate_mz_of_heavy_isotope_of_each_peptide(fdrDf):
+    return fdrDf.apply(
+        lambda x: calculate_mz_of_heavy_version_of_peptide(
+            x["peptide"], x["MzLIB"], x["zLIB"]
+        ),
+        axis=1,
+    )
+
+
+def make_bin_assignments_for_mz_values(mzValues, binWidth=0.75):
+    bins = np.arange(
+        int(min(mzValues)) - 1, int(max(mzValues) + binWidth * 3) + 1, binWidth * 2
+    )
+    values = np.digitize(mzValues, bins)
+    mzBinValues = bins[values]
+    mzBinValues -= binWidth
+    return mzBinValues
```

Comment:
Something doesn't seem right here. Using `mzValues = np.arange(0,10)`, we get bin assignments of `-0.25, 1.25, 2.75, ..., 8.75`.  
Similarly, `mzValues = np.arange(0,1,0.1)` results in two bins: `-0.25, 1.25`. Or 1/3 of `binWidth` below the min and above the max of the input.

Tell me if I'm missing something, but I think that this:
```Python
min(mzValues)-binWidth, max(mzValues) + binWidth * 3, binWidth * 2
```
gives the intended behaviour? The examples above map to `0, 1.5, 1.5, 3, 4.5, ...` and `0, [...], 1.5, 1.5`, which makes more sense to me.